ui/src/app/shared/services/auth-service.ts

```diff
@@ -1,8 +1,21 @@
+import { parse } from 'cookie';
+import * as jwt from 'jwt-decode';
+
 import { AuthSettings } from '../models';
 import requests from './requests';
 
 export class AuthService {
     public settings(): Promise<AuthSettings> {
         return requests.get('/settings').then((res) => res.body as AuthSettings);
     }
+
+    public getCurrentUserId(): string {
+        const cookies = parse(document.cookie);
+        const token = cookies['argocd.token'];
+        const user: any = token && jwt(token) || null;
+        if (user) {
+            return user.email || user.sub;
```

Comment:
should be anon?

Reply:
it makes sense. changed to return 'anonymous' is there is no user

Reply:
sorry - I meant to say we should obfuscate the UID before we send it to Google

Reply:
as we agreed offline, added setting to obfuscate user ids. By default user ids are obfuscated